dashboard/client/src/pages/job/hook/useJobProgress.ts

```diff
@@ -15,15 +15,23 @@ import { getJobProgress, getJobProgressByTaskName } from "../../../service/job";
 export const useJobProgress = (jobId?: string) => {
   const [msg, setMsg] = useState("Loading progress...");
   const [error, setError] = useState(false);
+  const [driverExits, setDriverExits] = useState(true);
```

Comment:
no need to use state for this. Instead, useSWR has a behavior to not fetch if the fetch key is null. 

So instaed of `useSWR("useJobProgress", async () => { ... }`, let's do `useSWR(jobId ? ["useJobProgress", jobId] : null, async () => { ... }`.

Also there's an existing bug, the jobId needs to be part of the fetchKey because we use it in the request body.

Then near the bottom, you can just return driverExists: false if jobId is undefined.

Reply:
I put up a PR for the existing bug fix:
https://github.com/ray-project/ray/pull/30495

Reply:
I will address the comment after merging that PR